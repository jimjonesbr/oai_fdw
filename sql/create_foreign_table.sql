CREATE FOREIGN TABLE ulb_ulbmsuo_oai_dc (
  id text                OPTIONS (oai_node 'identifier'),
  xmldoc xml             OPTIONS (oai_node 'content'),
  sets text[]            OPTIONS (oai_node 'setspec'),
  updatedate timestamp   OPTIONS (oai_node 'datestamp'),
  format text            OPTIONS (oai_node 'metadataprefix'),
  status boolean         OPTIONS (oai_node 'status')
 ) SERVER oai_server_ulb OPTIONS (setspec 'ulbmsuo', 
                                  metadataPrefix 'oai_dc');

SELECT * FROM ulb_ulbmsuo_oai_dc;

CREATE FOREIGN TABLE foreign_table_without_oai_option (
  foo text
 ) SERVER oai_server_ulb OPTIONS (setspec 'ulbmsuo',
                                  metadataPrefix 'oai_dc');
                                 
SELECT * FROM foreign_table_without_oai_option;

-- FOREIGN TABLE without 'oai_node' OPTION but with the 
-- mandatory SERVER OPTION nodes.
CREATE FOREIGN TABLE foreign_table_without_oai_option2 (
  foo text
 ) SERVER oai_server_ulb OPTIONS (metadataPrefix 'oai_dc');
 
SELECT * FROM foreign_table_without_oai_option2;
 