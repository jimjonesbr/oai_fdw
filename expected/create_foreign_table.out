CREATE FOREIGN TABLE ulb_ulbmsuo_oai_dc (
  id text                OPTIONS (oai_node 'identifier'),
  xmldoc xml             OPTIONS (oai_node 'content'),
  sets text[]            OPTIONS (oai_node 'setspec'),
  updatedate timestamp   OPTIONS (oai_node 'datestamp'),
  format text            OPTIONS (oai_node 'metadataprefix'),
  status boolean         OPTIONS (oai_node 'status')
 ) SERVER oai_server_ulb OPTIONS (setspec 'ulbmsuo', 
                                  metadataPrefix 'oai_dc');
CREATE TABLE tb_ulb_ulbmsuo_oai_dc AS SELECT * FROM ulb_ulbmsuo_oai_dc;
SELECT
  count(*) = count(distinct id) records_are_distinct,
  count(*) > 0 AS not_empty
FROM tb_ulb_ulbmsuo_oai_dc;
 records_are_distinct | not_empty 
----------------------+-----------
 t                    | t
(1 row)

CREATE FOREIGN TABLE foreign_table_without_oai_option (
  foo text
 ) SERVER oai_server_ulb OPTIONS (setspec 'ulbmsuo',
                                  metadataPrefix 'oai_dc');
                                 
SELECT * FROM foreign_table_without_oai_option;
 foo 
-----
(0 rows)

-- FOREIGN TABLE without 'oai_node' OPTION but with the 
-- mandatory SERVER OPTION nodes.
CREATE FOREIGN TABLE foreign_table_without_oai_option2 (
  foo text
 ) SERVER oai_server_ulb OPTIONS (metadataPrefix 'oai_dc');
 
SELECT * FROM foreign_table_without_oai_option2;
 foo 
-----
(0 rows)

 
